x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/persistence/DatafeedConfigProvider.java

```diff
@@ -197,26 +270,19 @@ public void onResponse(GetResponse getResponse) {
                     return;
                 }
 
-                try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
-                    XContentBuilder updatedSource = updatedConfig.toXContent(builder, ToXContent.EMPTY_PARAMS);
-                    IndexRequest indexRequest = client.prepareIndex(AnomalyDetectorsIndex.configIndexName(),
-                            ElasticsearchMappings.DOC_TYPE, DatafeedConfig.documentId(updatedConfig.getId()))
-                            .setSource(updatedSource)
-                            .setVersion(version)
-                            .request();
-
-                    executeAsyncWithOrigin(client, ML_ORIGIN, IndexAction.INSTANCE, indexRequest, ActionListener.wrap(
-                            indexResponse -> {
-                                assert indexResponse.getResult() == DocWriteResponse.Result.UPDATED;
-                                updatedConfigListener.onResponse(updatedConfig);
-                            },
-                            updatedConfigListener::onFailure
-                    ));
+                ActionListener<Boolean> validatedListener = ActionListener.wrap(
+                        ok -> {
+                            indexUpdatedConfig(updatedConfig, version, ActionListener.wrap(
+                                    indexResponse -> {
+                                        assert indexResponse.getResult() == DocWriteResponse.Result.UPDATED;
+                                        updatedConfigListener.onResponse(updatedConfig);
+                                    },
+                                    updatedConfigListener::onFailure));
+                        },
+                        updatedConfigListener::onFailure
+                );
 
-                } catch (IOException e) {
-                    updatedConfigListener.onFailure(
-                            new ElasticsearchParseException("Failed to serialise datafeed config with id [" + datafeedId + "]", e));
-                }
+                validator.accept(updatedConfig, validatedListener);
```

Comment:
When the datafeed is updated it must be validated against the job. We don't have the job here so the idea behind this call is go and validate against the job. See `TransportUpdateDatafeedAction.validateDatafeedJob`